jina/jaml/parsers/executor/legacy.py

```diff
@@ -12,14 +12,14 @@ class LegacyParser(VersionedYAMLParser):
     @staticmethod
     def _get_dump_path_from_config(meta_config: Dict):
         if 'name' in meta_config:
-            if meta_config.get('separated_workspace', False) is True and meta_config['pea_id'] != -1:
+            if meta_config['pea_id'] != -1:
                 if 'pea_id' in meta_config and isinstance(meta_config['pea_id'], int):
                     work_dir = meta_config['pea_workspace']
                     dump_path = os.path.join(work_dir, f'{meta_config["name"]}.{"bin"}')
                     if os.path.exists(dump_path):
                         return dump_path
                 else:
-                    raise BadWorkspace('separated_workspace=True but pea_id is unset or set to a bad value')
+                    raise BadWorkspace('pea_id is unset or set to a bad value')
```

Comment:
Bad error message, may u say that the path does not exist?

Reply:
I think the message wants to say that the pea_id is not set to a value the path creation can work with.
However, I think for this pr we should just remove the separated_workspace attribute.

Reply:
pea_id is unset or is not an integer is the condition, but I feel this condition was there to support the case where `separated_workspace` was True but `pea_id` was not coherent.